libbeat/tests/compose/wrapper.go

```diff
@@ -95,7 +98,47 @@ func (d *wrapperDriver) Up(ctx context.Context, opts UpOptions, service string)
 		args = append(args, service)
 	}
 
-	return d.cmd(ctx, "up", args...).Run()
+	var stdout, stderr bytes.Buffer
+	defer io.Copy(os.Stdout, &stdout)
+	defer io.Copy(os.Stderr, &stderr)
+	for {
+		// Up can fail if we have reached some system limit, specially
+		// number of networks, retry while the context is not done
+		cmd := d.cmd(ctx, "up", args...)
+		cmd.Stdout = &stdout
+		cmd.Stderr = &stderr
+
+		err := cmd.Run()
+		if err == nil {
+			return nil
+		}
+		if err := fatalError(&stderr); err != nil {
+			return errors.Wrapf(err, "docker-compose up failed for service %s", service)
+		}
+
+		select {
+		case <-time.After(time.Second):
+		case <-ctx.Done():
+			return err
+		}
+	}
+}
+
+var recoverableErrors = []string{
+	`could not find an available, non-overlapping IPv4 address pool`,
```

Comment:
is this that common? I wonder if this should be fatal (something to handle in CI)

Reply:
It happens if multiple scenarios are started at the same time, what can happen if we merge the parts of #7957 for multiple docker compose scenarios. I am going to close this till this happen.